src/arch/x86/analyzex86.rs

```diff
@@ -454,27 +423,34 @@ pub struct MemoryManager<'a>
     pub list: Vec<MemoryBounds<'a>>,
 }
 
-fn read_int<T: Num + BitOr<Output=T> + NumCast + Copy>(
-     _byte: &mut T,
```

Comment:
I think one of the more important changes, worth applying elsewhere if you agree, is to avoid the out-parameter argument passing style, especially if there's only one and the return value is `bool`. In those cases it's almost certain you can return `Option<T>` instead, which compiles to approximately the same code, but makes call sites less noisy and makes for fewer inputs to think about when reading code or writing tests.